shared/metrics/src/main/java/io/pravega/shared/metrics/StatsProviderImpl.java

```diff
@@ -47,17 +40,10 @@
 
     @Synchronized
     private void init() {
-        // I'm not entirely sure that re-inserting is necessary, but given that
-        // at this point we are preserving the registry, it seems safer to remove
-        // and re-insert.
-        MemoryUsageGaugeSet memoryGaugeNames = new MemoryUsageGaugeSet();
-        GarbageCollectorMetricSet gcMetricSet = new GarbageCollectorMetricSet();
-
-        memoryGaugeNames.getMetrics().forEach((key, value) -> metrics.remove(key));
-        gcMetricSet.getMetrics().forEach((key, value) -> metrics.remove(key));
-
-        metrics.registerAll(new MemoryUsageGaugeSet());
-        metrics.registerAll(new GarbageCollectorMetricSet());
+        new JvmMemoryMetrics().bindTo(metrics);
```

Comment:
What are this dangling objects? Do they require cleanup?

Reply:
This is the way Micrometer binds system metrics, per the documents.